### DISCUSSION TITLE
Run isolation when materializing multiple assets

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18176] [Q&A] [answered]
Hello. I am currently deploying Dagster against my own Kubernetes cluster (AWS EKS). When I run a job that contains multiple assets materialization, or I manually materialize more than one assets in the UI, I noticed it only create a single container: `dagster-run-xxx` to run all the asset materialization:
• Does anyone know of a way to match 1-to-1 asset to its corresponding container?
• Any feedback on best practices, guidance, etc. would be appreciated as well

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1700517728749529?thread_ts=1700517728.749529&cid=C01U954MEER).*

**Comment by benpankow** [comment, marked as answer]
By default, Dagster runs will execute in the context of a single container if deployed on Kubernetes. When materializing many assets in a single run, only one run container will generally be created.

For information on running each asset or compute step in its own container, see #14387.
